app/src/main/java/app/f3d/F3D/android/MainView.java

```diff
@@ -67,11 +71,22 @@ public void onSurfaceCreated(GL10 gl, EGLConfig config) {
             MainView.this.mEngine.getOptions().toggle("render.background.skybox");
             MainView.this.mEngine.getOptions().toggle("ui.filename");
             MainView.this.mEngine.getOptions().toggle("ui.loader-progress");
-
-            // hard-coded path, change it
-            MainView.this.mEngine.getLoader().loadScene("/data/local/tmp/WaterBottle.glb");
+//            MainView.this.mEngine.getOptions().toggle("model.volume.enable");
+
+            if(!Objects.equals(internalCachePath, "")) {
+                if(useGeometryFlag){
+                    MainView.this.mEngine.getLoader().loadGeometry(internalCachePath);
+                }else{
+                    MainView.this.mEngine.getLoader().loadScene(internalCachePath);
+                }
+            }
         }
     }
+    public void updateFilePath(String newFilePath, boolean useGeometry) {
+        // Use the new file path as needed in MainView
+        internalCachePath = newFilePath;

```

Comment:
Since `internalCachePath` and `useGeometryFlag` are only used in `onSurfaceCreated` (which is called only once at the creation of the view, how is it updated?

Reply:
A good catch here,

So when we open file explorer, the activity pauses and when we are back it recreates the mainview.
So the new path is used for rendering.

Im also not sure will recreating activity like this have any adverse affects need to check this out

Reply:
Just cross referenced from chat gpt and android life cycle doc's.
So opening an new intent pauses and recreates the activity , so the new path is being used.

Reply:
For the the time being I think it's good enough, but ideally we should try to avoid building the view again because initializing the OpenGL context could be expensive.